src/list.rs

```diff
@@ -105,7 +105,7 @@ impl Client {
     pub(crate) async fn list(&self, path: &Path, offset: Option<&Path>) -> crate::Result<Vec<ObjectMeta>> {
         with_retries!(self, self.list_impl(path, offset).await)
     }
-    async fn list_stream_impl(&self, prefix: &Path, offset: Option<&Path>) -> crate::Result<ListStream> {
+    pub(crate) async fn list_stream_impl(&self, prefix: &Path, offset: Option<&Path>) -> crate::Result<ListStream> {
```

Comment:
do we need to declare it `pub(crate)` because it's used by `pub(crate)` `download_prefix_impl`?

Reply:
Correct, we make it `pub(crate)` so that we can use within `download_prefix_impl`. There are still places where the visibility is too restrictive but I will be relaxing it on-demand